File: backend/src/main/java/com/ams/entity/BpmMailConfig.java
package com.ams.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDateTime;

@Data
@TableName("bpm_mail_config")
public class BpmMailConfig implements Serializable {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String processType;

    private String processName;

    private String nodeId;

    private String nodeName;

    private String subjectTemplate;

    private String contentTemplate;

    private String toRecipients;

    private String ccRecipients;

    private Integer enabled;

    private String remark;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}

File: frontend/src/types/bpmMailConfig.ts
export interface BpmMailConfig {
  id: number;
  processType: string;
  processName?: string;
  nodeId?: string;
  nodeName?: string;
  subjectTemplate?: string;
  contentTemplate?: string;
  toRecipients?: string;
  ccRecipients?: string;
  enabled: number;
  remark?: string;
  createdAt?: string;
  updatedAt?: string;
}

export interface BpmMailVariable {
  id: number;
  varKey: string;
  varName: string;
  defaultValue?: string;
  remark?: string;
  createdAt?: string;
  updatedAt?: string;
}

export interface CreateBpmMailConfigRequest {
  processType: string;
  processName?: string;
  nodeId?: string;
  nodeName?: string;
  subjectTemplate?: string;
  contentTemplate?: string;
  toRecipients?: string;
  ccRecipients?: string;
  enabled?: number;
  remark?: string;
}

export interface UpdateBpmMailConfigRequest {
  processName?: string;
  nodeId?: string;
  nodeName?: string;
  subjectTemplate?: string;
  contentTemplate?: string;
  toRecipients?: string;
  ccRecipients?: string;
  enabled?: number;
  remark?: string;
}

export interface CreateBpmMailVariableRequest {
  varKey: string;
  varName: string;
  defaultValue?: string;
  remark?: string;
}

export interface UpdateBpmMailVariableRequest {
  varName?: string;
  defaultValue?: string;
  remark?: string;
}

export const PROCESS_TYPES: Record<string, string> = {
  RETIREMENT: '报废流程',
  WORK_ORDER: '工单流程',
  PURCHASE: '采购流程',
  MAINTENANCE: '维保流程',
  INVENTORY: '盘点流程',
};
